runtime/src/main/java/org/corfudb/runtime/collections/StreamListenerResumePolicy.java

```diff
@@ -0,0 +1,103 @@
+package org.corfudb.runtime.collections;
+
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.CorfuStoreMetadata.Timestamp;
+import org.corfudb.runtime.exceptions.StreamingException;
+import org.corfudb.runtime.exceptions.UnreachableClusterException;
+import org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuInterruptedError;
+import org.corfudb.util.retry.IRetry;
+import org.corfudb.util.retry.IntervalRetry;
+import org.corfudb.util.retry.RetryNeededException;
+
+import java.util.List;
+
+/**
+ * This is an abstract stream listener callback implementation for clients interested in automatic
+ * re-subscription upon any error. The re-subscription policy of this implementation protects against data
+ * loss by immediately performing a full sync on any type of error.
+ *
+ * Created by annym on 06-22-2021
+ */
+@Slf4j
+public abstract class StreamListenerResumePolicy implements StreamListener {
```

Comment:
This policy class seems to be a base class. Are we going to expose it to users? If not, might be good to make it package private?

Reply:
correct, I started off making it a class to expose to users but it ended up as a base class. Making it private.